### PR TITLE
chore: add assocs function to Type.scala

### PR DESCRIPTION
**Pull request by mlutze**
extracted from #6499

### CODE REVIEW

**Review by JonathanStarup — CHANGES_REQUESTED** [review]
For demo

**Review by JonathanStarup — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/language/ast/Type.scala [hunk 1]
```diff
@@ -83,6 +83,19 @@ sealed trait Type {
     case Type.AssocType(_, arg, _, _) => arg.cases // TODO ASSOC-TYPES throw error?
   }
 
+  /**
+    * Gets all the associated types in the given type.
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/ast/Type.scala:87]: Returns, but okay

File: main/src/ca/uwaterloo/flix/language/ast/Type.scala [hunk 2]
```diff
@@ -83,6 +83,19 @@ sealed trait Type {
...
+    */
+  def assocs: Set[Type.AssocType] = this match {
+    case t: Type.AssocType => Set(t)
+
+    case _: Type.Cst => Set.empty
+    case _: Type.Var => Set.empty
+
+    case Type.Apply(tpe1, tpe2, _) => tpe1.assocs ++ tpe2.assocs
+    case Type.Alias(_, _, tpe, _) => tpe.assocs
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/ast/Type.scala:96]: Above not in the right order :)

**Review by magnus-madsen — APPROVED** [review]

**Comment by magnus-madsen** [comment]
@mlutze Approved, but you can beautify or merge. I leave it to you.

**Pull request merged**